tests/infer/test_enum.py

```diff
@@ -169,18 +175,89 @@ def test_gmm_batch_iter_discrete_traces(model, data_size, graph_type):
     (gmm_model, gmm_guide),
     (gmm_batch_model, gmm_batch_guide),
 ], ids=["single", "batch"])
-@pytest.mark.parametrize("enum_discrete", [False, True], ids=["sample", "sum"])
+@pytest.mark.parametrize("enum_discrete", [None, "sequential", "parallel"])
 def test_svi_step_smoke(model, guide, enum_discrete, trace_graph):
     pyro.clear_param_store()
     data = Variable(torch.Tensor([0, 1, 9]))
 
+    guide = config_enumerate(guide, default=enum_discrete)
     optimizer = pyro.optim.Adam({"lr": .001})
     inference = SVI(model, guide, optimizer, loss="ELBO",
-                    trace_graph=trace_graph, enum_discrete=enum_discrete)
+                    trace_graph=trace_graph, max_iarange_nesting=1)
     with xfail_if_not_implemented():
         inference.step(data)
 
 
+@pytest.mark.parametrize("enum_discrete", [None, "sequential", "parallel"])
+@pytest.mark.parametrize("trace_graph", [False, True], ids=["dense", "flat"])
+def test_bern_elbo_gradient(enum_discrete, trace_graph):
```

Comment:
FYI This and the following test use analytic `grad(kl_divergence(-, -), -)` to test gradients.